task_processing/task_processing/execution_framework.py

```diff
@@ -395,3 +338,222 @@ def unblacklist_slave(self, slave_id):
+                volume = container.volumes.add()
+                volume.container_path = container_path
+                volume.host_path = host_path
+                """
+                volume.mode = 1 # mesos_pb2.Volume.Mode.RW
+                volume.mode = 2 # mesos_pb2.Volume.Mode.RO
+                """
+                volume.mode = 1 if mode == "RW" else 2
+
+        # Container info
+        docker = mesos_pb2.ContainerInfo.DockerInfo()
+        docker.image = self.task_config.image
+        docker.network = 2  # mesos_pb2.ContainerInfo.DockerInfo.Network.BRIDGE
+        docker.force_pull_image = True
+
+        available_ports = []
+        for resource in offer.resources:
+            if resource.name == "ports":
+                available_ports = self.get_available_ports(resource)
+
+        port_to_use = available_ports[0]
+
+        mesos_ports = task.resources.add()
+        mesos_ports.name = "ports"
+        mesos_ports.type = mesos_pb2.Value.RANGES
+        port_range = mesos_ports.ranges.range.add()
+
+        port_range.begin = port_to_use
+        port_range.end = port_to_use
+        docker_port = docker.port_mappings.add()
+        docker_port.host_port = port_to_use
+        docker_port.container_port = 8888
+
+        # Set docker info in container.docker
+        container.docker.MergeFrom(docker)
+        # Set docker container in task.container
+        task.container.MergeFrom(container)
+
+        return task
+
+    ####################################################################
+    #                   Mesos driver hooks go here                     #
+    ####################################################################
+
+    def slaveLost(self, drive, slaveId):
+        log.error("Slave lost: {id}".format(id=str(slaveId)))
+
+    def registered(self, driver, frameworkId, masterInfo):
+        self.framework_id = frameworkId.value
+        paasta_print("Registered with framework ID %s" % frameworkId.value)
+
+        self.reconcile_start_time = time.time()
+        driver.reconcileTasks([])
+
+    def resourceOffers(self, driver, offers):
+        if self.frozen:
+            return
+
+        if self.within_reconcile_backoff():
+            paasta_print("Declining all offers since we started reconciliation too recently")
+            for offer in offers:
+                driver.declineOffer(offer.id)
+        else:
+            for idx, offer in enumerate(offers):
+                if offer.slave_id.value in self.blacklisted_slaves:
+                    log.critical("Ignoring offer %s from blacklisted slave %s" %
+                                 (offer.id.value, offer.slave_id.value))
+                    driver.declineOffer(offer.id)
+                    del offers[idx]
+
+            if len(offers) == 0:
+                return
+
+            self.launch_tasks_for_offers(driver, offers)
+
+    def statusUpdate(self, driver, update):
```

Comment:
sure. i just didn't want to invest time into something that we might choose not to merge.